lib/spack/spack/fetch_strategy.py

```diff
@@ -890,8 +890,17 @@ def source_id(self):
 
     def mirror_id(self):
         if self.commit:
+            provenance_id = self.commit
             repo_path = urllib.parse.urlparse(self.url).path
-            result = os.path.sep.join(["git", repo_path, self.commit])
+            if self.git_sparse_paths:
+                sparse_paths = []
+                if callable(self.git_sparse_paths):
+                    sparse_paths.extend(self.git_sparse_paths())
+                else:
+                    sparse_paths.extend(self.git_sparse_paths)
+                sparse_hash = hash(tuple(sparse_paths))
```

Comment:
you'll have to use some algorithm from hashlib, `hash` depends on the python interpreter.